### ISSUE TITLE
Target all JavaScript platforms with one generator

### ISSUE THREAD

**Issue by lucat1** [apex/rpc#15]
## Prerequisites

* [X] I searched to see if the issue already exists.
Related to #7 .

## Description

The library could be made isomorphic and therefore also run on client-side JavaScript runtimes if we used a library like `isomorphic-(un)fetch`. Furthermore using a cross-environment fetch implementation would work on both client-side and deno.
Node would still need to import a package AFAIK ~you could define the function in the global scope outside of this library, before requiring it, but I don't think it would be ideal.~

To make the node implementation isomorphic it would take nothing more than changing the `node-fetch` import to some isomorphic library. I suggest we do this change internally rather than relying on the user aliasing the module with some bundler like webpack, as not everyone uses it. 

Keeping all of this in mind, I think we could could kill two birds with one stone here by simply adding a `--fetch-library` argument to specify the desired input library or removing it (for deno) with some arbitrary value.

Examples:
`--fetch-library=isomorphic-unfetch`:
```js
import fetch from 'isomorphic-unfetch'
```

`--fetch-library=deno`:
```js
/* no import required */
```

**Comment by lucat1** [comment]
OT: supposing the generated code will be used on the client side, it would benefit A LOT if we exported single functions rather than a class with a bunch of methods as the former would allow for tree-shaking and dead-code elimination.

**Comment by tj** [comment]
SGTM, it would definitely help me as well if they were maintained as one in https://github.com/apex/logs/tree/master/js

I think a class is still appropriate though, it would be a bit awkward to keep passing in the `url` and `authToken` etc

**Comment by lucat1** [comment]
Yup, I totally get why you used a class instead of a bunch of functions, but this approach isn't currently tree shakeable(not a huge deal unless you have a really big schema).

This should be fairly simple to implement, just an if statement [here](https://github.com/apex/rpc/blob/master/generators/tsclient/tsclient.go#L48).
I can write a PR for this if we're okay with the argument name.

**Comment by tj** [comment]
Awesome sounds good!

I'm not too worried about DCE for now, my JS bundle is like 3.5mb uncompressed hahah, the API portion is a tiny portion. I'd love to keep all the clients using a similar look/feel as long as it makes sense for the language, there could always be a third-party generator which outputs functions

**Comment by lucat1** [comment]
Should be fixed with #18
